src/common/utils/http.ts

```diff
@@ -66,21 +67,36 @@ export async function httpGet<TBody extends JsonMap | string = string , TData =
             encoding: undefined,
             headers,
         },
-    ) as TRequestCoreOptionsRequiredUriUrl;
+    );
 
-    const promisifiedRequest = promisify<TRequestCoreOptionsRequiredUriUrl, request.Response>(request);
-    const response = await promisifiedRequest(requestOptions);
-
-    const result = {
-        isFailure: response.statusCode < 200 || response.statusCode >= 300,
-        isSuccess: response.statusCode >= 200 && response.statusCode < 300,
-        url,
-        responseUrl: response.url,
-        statusCode: response.statusCode,
-        body: response.body,
-        data: callback ? undefined : response.body,
-        contentType: response.caseless.get("Content-Type"),
-    };
+    const result = await (async (opt): Promise<IHttpGetResult<TBody, TData>> =>
+        new Promise((resolve, reject) => {
+            request(opt, (err, response) => {
+                if (err) {
+                    if (err.code === "ETIMEDOUT") {
+                        resolve({
+                            isTimeout: true,
```

Comment:
Now that the `ETIMEDOUT` error is captured "internally" and transformed into a promise.resolve (it was a promise.reject before, only optionally captured by the caller of `httpGet()`), there is a loss of information, namely the `err.connect` boolean. I therefore suggest that this information be encoded as `timeoutConnect: boolean` in the `IHttpGetResult<TBody, TData>`.
https://github.com/request/request#timeouts

Reply:
done